core/templates/pages/email-dashboard-pages/email-dashboard-page.controller.spec.ts

```diff
@@ -0,0 +1,207 @@
+      $rootScope: $scope
+    });
+
+    spyOn(EmailDashboardDataService, 'isNextPageAvailable').and.returnValue(
+      true);
+    spyOn(EmailDashboardDataService, 'isPreviousPageAvailable').and.returnValue(
+      true);
+
+    spyOn(UserService, 'getUserInfoAsync').and.callFake(function() {
+      var deferred = $q.defer();
+      deferred.resolve({
+        getUsername: function() {
+          return 'username';
+        }
+      });
+      return deferred.promise;
+    });
+  }));
+
+  it('should reset form', function() {
+    // Mock some values.
+    ctrl.has_not_logged_in_for_n_days = true;
+    ctrl.inactive_in_last_n_days = true;
+    ctrl.created_at_least_n_exps = true;
+    ctrl.created_fewer_than_n_exps = false;
+    ctrl.edited_at_least_n_exps = true;
+    ctrl.edited_fewer_than_n_exps = false;
+
+    ctrl.resetForm();
+
+    expect(ctrl.has_not_logged_in_for_n_days).toBe(null);
+    expect(ctrl.inactive_in_last_n_days).toBe(null);
+    expect(ctrl.created_at_least_n_exps).toBe(null);
+    expect(ctrl.created_fewer_than_n_exps).toBe(null);
+    expect(ctrl.edited_at_least_n_exps).toBe(null);
+    expect(ctrl.edited_fewer_than_n_exps).toBe(null);
+  });
+
+  it('should submit query', function() {
+    // Mock some values.
+    ctrl.has_not_logged_in_for_n_days = true;
+    ctrl.inactive_in_last_n_days = true;
+    ctrl.created_at_least_n_exps = true;
+    ctrl.created_fewer_than_n_exps = false;
+    ctrl.edited_at_least_n_exps = true;
+    ctrl.edited_fewer_than_n_exps = false;
+
+    spyOn(EmailDashboardDataService, 'submitQuery').and.callFake(function() {
+      var deferred = $q.defer();
+      deferred.resolve(firstPageQueries);
+      return deferred.promise;
+    });
+
+    ctrl.submitQuery();
+    $scope.$apply();
+
+    expect(ctrl.currentPageOfQueries).toEqual(firstPageQueries);
+    expect(ctrl.showSuccessMessage).toBe(true);
+
+    expect(ctrl.has_not_logged_in_for_n_days).toBe(null);
+    expect(ctrl.inactive_in_last_n_days).toBe(null);
+    expect(ctrl.created_at_least_n_exps).toBe(null);
+    expect(ctrl.created_fewer_than_n_exps).toBe(null);
+    expect(ctrl.edited_at_least_n_exps).toBe(null);
+    expect(ctrl.edited_fewer_than_n_exps).toBe(null);
+  });
+
+  it('should get next page of queries', function() {
+    spyOn(EmailDashboardDataService, 'getNextQueries').and.callFake(
+      function() {
+        var deferred = $q.defer();
+        deferred.resolve(secondPageQueries);
+        return deferred.promise;
+      });
+
+    ctrl.getNextPageOfQueries();
+    $scope.$apply();
+
+    expect(ctrl.currentPageOfQueries).toEqual(secondPageQueries);
+  });
+
+  it('should get previous page of queries', function() {
+    spyOn(EmailDashboardDataService, 'getPreviousQueries').and.returnValue(
+      firstPageQueries);
+
+    ctrl.getPreviousPageOfQueries();
+
+    expect(ctrl.currentPageOfQueries).toEqual(firstPageQueries);
+  });
+
+
+  it('should get previous page of queries', function() {
```

Comment:
Doesn't seem like it's "getting" anything, from the contents of this test...

Ditto below. Please check that your test names accurately describe what the test is doing.

Reply:
Done.